src/eos-update-server.c

```diff
@@ -263,6 +264,58 @@ data_generate_config (Data *data,
   return TRUE;
 }
 
+static gchar *
+quit_file_name (void)
+{
+  return eos_updater_dup_envvar_or ("EOS_UPDATER_TEST_UPDATE_SERVER_QUIT_FILE",
+                                     NULL);
+}
+
+static EosQuitFileCheckResult
+check_and_quit (gpointer data_ptr)
+{
+  Data *data = data_ptr;
+  FILE *log;
+  g_autofree gchar *str = NULL;
+
+  log = fopen ("eos-update-server.log", "a");
+  str = g_strdup_printf ("requests pending: %u\n",
+                         data->async_requests_pending);
+  fwrite (str, 1, strlen (str), log);
```

Comment:
Gah, this code should not be here. I added it temporarily and apparently forgot to remove it.